src/shogun/machine/RandomForest.cpp

```diff
@@ -159,11 +159,21 @@ void CRandomForest::set_machine_parameters(CMachine* m, SGVector<index_t> idx)
 	}
 
 	tree->set_weights(weights);
-
+	tree->set_sorted_features(m_sorted_feats, m_sorted_indices);
 	// equate the machine problem types - cloning does not do this
 	tree->set_machine_problem_type(dynamic_cast<CRandomCARTree*>(m_machine)->get_machine_problem_type());
 }
 
+bool CRandomForest::train_machine(CFeatures* data)
+{
+	SGMatrix<float64_t> mat=(dynamic_cast<CDenseFeatures<float64_t>*>(data))->get_feature_matrix();
```

Comment:
yep yep we've talked about this with @Saurabh7 
we should finish up with this PR as a performance fix, and then work with further improving trees/forests.